File: ios/NavigationBarManager.m
//
//  NavigationBarManager.m
//  stripe-react-native
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(NavigationBarManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(title, NSString)
RCT_EXPORT_VIEW_PROPERTY(onCloseButtonPress, RCTDirectEventBlock)
@end

File: ios/NavigationBarManager.swift
//
//  NavigationBarManager.swift
//  stripe-react-native
//

import Foundation
import React

@objc(NavigationBarManager)
class NavigationBarManager: RCTViewManager {
    override func view() -> UIView! {
        return NavigationBarView()
    }

    override class func requiresMainQueueSetup() -> Bool {
        return true
    }
}
